### ISSUE TITLE
Exception handling

### ISSUE THREAD

**Issue by belltailjp** [belltailjp/chainer_slack_report#2]
In case an exception is raised during the training, SlackReport will eventually kill the whole training process, which is a nightmare for users.
First it to do as much error checking as possible in the initializer to grab errors before the training starts, and once the training starts, exceptions should be reinterpreted as warnings.

**Comment by belltailjp** [comment]
There was an example.
```
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='slack.com', port=443): Max retries exceeded with url: /api/chat.update (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8a500c5710>: Failed to establish a new connection: [Errno -2] Name or service not known'))
```

**Comment by belltailjp** [comment]
Solved by #5
